pulp_file/tests/functional/api/from_pulpcore/test_reclaim_disk_space.py

```diff
@@ -1,193 +1,170 @@
+        repo = file_repo_api_client.read(repo.pulp_href)
 
-        reclaim_response = self.reclaim_api.reclaim({"repo_hrefs": [repo.pulp_href]})
-        monitor_task(reclaim_response.task)
+        # Publication
+        publication_data = FileFilePublication(repository=repo.pulp_href)
+        publication = gen_object_with_cleanup(file_pub_api_client, publication_data)
 
-        artifacts_after_reclaim = self.artifacts_api.list().count
-        self.assertLess(artifacts_after_reclaim, artifacts_before_reclaim)
+        # Distribution
+        body = gen_distribution()
+        body["publication"] = publication.pulp_href
+        distribution = gen_object_with_cleanup(file_distro_api_client, body)
 
-        download_content_unit(self.cfg, distribution.to_dict(), content["relative_path"])
-        artifacts_after_download = self.artifacts_api.list().count
-        # Downloading a reclaimed content will increase the artifact count by 1
-        self.assertEqual(artifacts_after_download, artifacts_after_reclaim + 1)
-        # But only 1 extra artifact will be downloaded, so still less than after immediate sync
-        self.assertLess(artifacts_after_download, artifacts_before_reclaim)
+        return repo, distribution
 
-    def test_specified_all_repos(self):
-        """Tests that specifying all repos w/ '*' properly grabs all the repos."""
-        repos = [self.repo_api.create(gen_repo()) for _ in range(10)]
-        for repo in repos:
-            self.addCleanup(self.repo_api.delete, repo.pulp_href)
+    return _sync_repository_distribution
 
-        repos = [r.pulp_href for r in self.all_repo_api.list().results]
 
-        reclaim_response = self.reclaim_api.reclaim({"repo_hrefs": ["*"]})
-        task_status = monitor_task(reclaim_response.task)
+def test_reclaim_on_demand_content(
+    sync_repository_distribution,
+    artifacts_api_client,
+    repositories_reclaim_space_api_client,
+    download_content_unit,
+    delete_orphans_pre,
+):
+    """
+    Test whether on_demand repository content can be reclaimed
+    and then re-populated back after client request.
+    """
+    repo, distribution = sync_repository_distribution(policy="on_demand")
 
-        repos_locked = [r.split(":")[-1] for r in task_status.reserved_resources_record]
-        self.assertEqual(len(repos), len(repos_locked))
-        self.assertEqual(set(repos), set(repos_locked))
+    artifacts_before_download = artifacts_api_client.list().count
+    content = get_content(repo.to_dict())[FILE_CONTENT_NAME][0]
+    download_content_unit(distribution.base_path, content["relative_path"])
 
-    def _repo_sync_distribute(self, policy="immediate"):
-        """Helper to create & populate a repository and distribute it."""
-        repo = self.repo_api.create(gen_repo())
-        self.addCleanup(self.repo_api.delete, repo.pulp_href)
+    artifacts = artifacts_api_client.list().count
+    assert artifacts >= artifacts_before_download
```

Comment:
```suggestion
    assert artifacts > artifacts_before_download
```
The original assertion was just greater than.